### PR TITLE
Change accessibiilty on Payload.

### LINKED ISSUE #537: Payload class is internal - how then to perform request interception
### Description

I'm trying to set a http header for the initial request to a target webpage. I tried using `SetExtraHttpHeadersAsync` but that appears not to be it's intended use (ie it passes the header to *all* requests that the target web page is making)

I then tried to use request interception as per one of the [unit test](https://github.com/kblok/puppeteer-sharp/blob/aa6470f61ab167b1525ebb74468811f386736b79/lib/PuppeteerSharp.Tests/PageTests/SetRequestInterceptionTests.cs#L154). However I can't use this method due to the `Payload` class and it's `Headers` property being marked as internal.

Am I missing some other method of achieving this? Otherwise, can these symbols be made public?

Thanks
Christian

### PR DESCRIPTION
**Pull request by kblok**
As the Request class is never treated as a Payload we are removing that dependency.

closes #537

### CODE REVIEW

File: lib/PuppeteerSharp/Payload.cs [hunk 1]
```diff
@@ -23,67 +23,66 @@ internal Payload()
...
-                        {
-                            continue;
-                        }
-                        hash.Headers[item.Key] = item.Value;
-                    }
-                }
-
-                return JsonConvert.SerializeObject(hash);
-            }
+        internal string Hash
```
Comment (Meir017) [inline lib/PuppeteerSharp/Payload.cs]: Shouldn't this still have the JsonIgnore attribute?

Comment (kblok) [reply lib/PuppeteerSharp/Payload.cs]: Why would we need it if we are not serializing this object?

Comment (Meir017) [reply lib/PuppeteerSharp/Payload.cs]: In that case remove all of the attributes in this class

Comment (kblok) [reply lib/PuppeteerSharp/Payload.cs]: `JsonProperty` it's used for deserialization. e.g. https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp/Messaging/RequestInterceptedResponse.cs#L13

**Pull request merged**